src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java

```diff
@@ -54,26 +54,40 @@ public ActivityServiceImpl(ActivityRepository activityRepository, RecordsService
     @Override
     @Transactional(readOnly = true)
     public Page<ActivityDTO> getAllOrganizationActivities(Pageable pageable, UUID systemAccountId,
-        String search, SearchOn searchOn, FiltersActivityDTO filtersForActivity) {
-
-        List<ActivityDTO> activities = new ArrayList<>();
-        Page<ActivityInfo> activitiesInfo = findAllActivitiesInfoWithOwnerId(systemAccountId, search, searchOn, pageable,
-            filtersForActivity);
+        String search, SearchOn searchOn, FiltersActivityDTO filtersForActivity, Boolean showPartner) {
 
         Map<UUID, ExclusionsConfig> exclusionsMap = exclusionsConfigService.getAllBySystemAccountId();
 
-        for (ActivityInfo info : activitiesInfo) {
-            try {
-                activities.add(getEntityActivity(info, exclusionsMap));
-            } catch (ActivityCreationException ex) {
-                log.error(ex.getMessage());
+        List<ActivityDTO> activities = new ArrayList<>();
+        long totalElements;
+        if (showPartner) {
+            Page<ActivityInfo> partnerActivitiesInfo = findAllPartnerActivitiesInfo(systemAccountId, search, searchOn,
+                pageable, filtersForActivity);
+            totalElements = partnerActivitiesInfo.getTotalElements();
+            for (ActivityInfo info : partnerActivitiesInfo) {
+                try {
+                    activities.add(getEntityActivity(info, exclusionsMap));
+                } catch (ActivityCreationException ex) {
+                    log.error(ex.getMessage());
+                }
+            }
+        } else {
+            Page<ActivityInfo> activitiesInfo = findAllActivitiesInfoWithOwnerId(systemAccountId, search, searchOn, pageable,
```

Comment:
only this part is different, so you can extract the rest of this code outside the if

Reply:
done